[SofaGeneral] Pluginize SofaBoundaryCondition

fredroy opened the pull request:
#1529

A good ol' fat module.
A priori not breaking.

And usual cleaning (namespaces, pragma), 

~~This PR includes #1570 so this one must wait until it is merged~~ 
______________________________________________________
<!--- Please leave this at the end of your message -->
This PR: 
- builds with SUCCESS for all platforms on the CI.
- does not generate new warnings.
- does not generate new unit test failures.
- does not generate new scene test failures.
- does not break API compatibility.
- is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

fredroy committed:
Start modularization (cmake & folders)

fredroy committed:
clean sources

fredroy committed:
Update ComponentChange

fredroy committed:
update scenes

fredroy committed:
Merge branch 'modularize_general_collision' of https://github.com/fredroy/sofa into modularize_boundary_condition

fredroy commented:
[ci-build][with-all-tests]

fredroy committed:
Merge branch 'modularize_general_collision' of https://github.com/fredroy/sofa into modularize_boundary_condition

fredroy committed:
Merge branch 'master' of https://github.com/sofa-framework/sofa into modularize_boundary_condition

guparan committed:
Merge remote-tracking branch 'sofa-framework/master' into modularize_boundary_condition

fredroy committed:
Fix deps and cmake.in for out of tree compilations

fredroy committed:
Merge branch 'master' of https://github.com/sofa-framework/sofa into fix_outoftree_compil

guparan committed:
[CMake] CLEAN SofaMisc*Config.cmake.in

guparan committed:
[CMake] CLEAN SofaAdvanced*Config.cmake.in

guparan committed:
[CMake] CLEAN SofaGeneralConfig.cmake.in and SofaGuiConfig.cmake.in

guparan committed:
[CMake] Use QUIET REQUIRED in *Config.cmake.in

guparan committed:
[CMake] FIX check_required_components calls

guparan committed:
[CMake] CLEAN variable naming for pluginization

guparan commented on tests/CMakeLists.txt:
```diff
@@ -39,7 +39,7 @@ add_subdirectory(${SOFA_KERNEL_MODULES_SOURCE_DIR}/SofaRigid/SofaRigid_test ${SO
 
 # SofaGeneral tests
-add_subdirectory(${SOFA_EXT_MODULES_SOURCE_DIR}/SofaBoundaryCondition/SofaBoundaryCondition_test ${SOFA_EXT_MODULES_BINARY_DIR}/SofaGeneral/SofaBoundaryCondition/SofaBoundaryCondition_test)
+#add_subdirectory(${SOFA_EXT_MODULES_SOURCE_DIR}/SofaBoundaryCondition/SofaBoundaryCondition_test ${SOFA_EXT_MODULES_BINARY_DIR}/SofaGeneral/SofaBoundaryCondition/SofaBoundaryCondition_test)
```
To be removed ?

fredroy replied:
yes

guparan commented on modules/SofaBoundaryCondition/CMakeLists.txt:
```diff
@@ -1,131 +1,146 @@
 cmake_minimum_required(VERSION 3.12)
-project(SofaBoundaryCondition)
+project(SofaBoundaryCondition VERSION 1.0 LANGUAGES CXX)
```
Do we want modules to be independently versioned instead of following SOFA versions?
If no version is given, Sofa_VERSION is set by sofa_auto_set_target_properties

fredroy replied:
1- Dont know for the version (something to discuss I suppose)
2- anyway, if I dont put a version there is an error while doing cmake's config:
```
CMake Error at C:/Program Files/CMake/share/cmake-3.18/Modules/WriteBasicConfigVersionFile.cmake:43 (message):
  No VERSION specified for WRITE_BASIC_CONFIG_VERSION_FILE()
```

guparan replied:
--> fixed in #1570

guparan committed:
[CMake] Uniform version for modules

Following Sofa_VERSION

guparan committed:
[SofaMacros] CLEAN version handling for targets

Target version will be:
  TargetName_VERSION if any (eq PROJECT_VERSION)
  - OR -
  PACKAGE_VERSION if any
  - OR -
  Sofa_VERSION if any

fredroy committed:
Merge branch 'fix_outoftree_compil' of https://github.com/fredroy/sofa into modularize_boundary_condition

guparan committed:
[SofaMacros] FIX missing PROJECT_VERSION

fredroy committed:
Merge branch 'fix_outoftree_compil' of https://github.com/fredroy/sofa into modularize_boundary_condition

fredroy committed:
fix version

fredroy commented:
[ci-build][force-full-build]

fredroy commented:
[ci-build][with-all-tests]

guparan committed:
Merge remote-tracking branch 'sofa-framework/master' into modularize_boundary_condition
